app/models/open_access_location.rb

```diff
@@ -1,24 +1,26 @@
 # frozen_string_literal: true
 
 class OpenAccessLocation < ApplicationRecord
-  def self.sources
-    [
-      'User',
-      'ScholarSphere',
-      'Open Access Button',
-      'Unpaywall',
-      'Dickinson Law IDEAS Repo',
-      'Penn State Law eLibrary Repo'
-    ]
-  end
+  enum source: to_enum_hash([
+                              Source::USER,
+                              Source::SCHOLARSPHERE,
+                              Source::OPEN_ACCESS_BUTTON,
+                              Source::UNPAYWALL,
+                              Source::DICKINSON_IDEAS,
+                              Source::PSU_LAW_ELIBRARY
+                            ]),
+       _prefix: :source
```

Comment:
This looks good. What is _prefix and where is to_enum_hash coming from?

Reply:
I'm assuming it's the same as https://github.com/psu-libraries/scholarsphere/blob/develop/app/models/work.rb#L98

Reply:
oh, I see... it's in ApplicationRecord.